src/lib/comms/sol-coap.c

```diff
@@ -1669,9 +1682,9 @@ sol_coap_server_destroy(struct sol_coap_server *server)
     }
 
     SOL_PTR_VECTOR_FOREACH_REVERSE_IDX (&server->pending, reply, i) {
-        sol_ptr_vector_del(&server->pending, i);
```

Comment:
Again, as [above](https://github.com/solettaproject/soletta/pull/2059/files#r63861131), the order
  0. remove-from-vector
  0. call-external-callback

is being reversed by this patch.